lib/membrane_rtc_engine/endpoints/webrtc/simulcast_tee.ex

```diff
@@ -147,8 +149,52 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.SimulcastTee do
         _other -> []
       end)
 
-    start_timer = [start_timer: {:check_encoding_statuses, Time.seconds(1)}]
-    {{:ok, start_timer ++ caps}, state}
+    {{:ok, caps}, state}
+  end
+
+  @impl true
+  def handle_event(Pad.ref(:input, {_track_id, encoding}), event, ctx, state) do
+    {actions, state} = handle_input_event(event, encoding, ctx, state)
+    {{:ok, actions}, state}
+  end
+
+  @impl true
+  def handle_event(pad, event, ctx, state) do
+    super(pad, event, ctx, state)
+  end
+
+  defp handle_input_event(%TrackVariantPaused{}, encoding, ctx, state) do
+    new_state = update_in(state, [:inactive_encodings], &MapSet.put(&1, encoding))
+
+    new_state =
+      Enum.reduce(state.forwarders, new_state, fn {endpoint_id, forwarder}, new_state ->
+        put_in(
+          new_state,
+          [:forwarders, endpoint_id],
+          Forwarder.encoding_inactive(forwarder, encoding)
+        )
+      end)
+
+    actions = generate_keyframe_requests(new_state, state, ctx)
+
+    {actions, new_state}
+  end
+
+  defp handle_input_event(%TrackVariantResumed{}, encoding, ctx, state) do
+    new_state = update_in(state, [:inactive_encodings], &MapSet.delete(&1, encoding))
```

Comment:
Why did you create additional private function instead of multiple handle_event clauses?